app/src/main/java/org/p2p/wallet/receive/token/ReceiveTokenPresenter.kt

```diff
@@ -54,12 +54,18 @@ class ReceiveTokenPresenter(
         }
     }
 
-    override fun saveQr(name: String, bitmap: Bitmap) {
+    override fun saveQr(name: String, bitmap: Bitmap, shareAfter: Boolean) {
         launch {
             try {
-                usernameInteractor.saveQr(name, bitmap)
-                view?.showToastMessage(R.string.auth_saved)
-            } catch (e: Exception) {
+                val savedFile = usernameInteractor.saveQr(name, bitmap)
```

Comment:
damn that's a lot of duplication
how about to extract sharing logic into a separate class? im not sure how tho

Reply:
i also thought about it
but... do not know how to make it in better way